### ISSUE TITLE
[GitHub] Unread notifications list clearing temporarily

### ISSUE THREAD

**Issue by asportnoy** [raycast/extensions#13023]
### Extension

https://www.raycast.com/raycast/github

### Raycast Version

1.76.1

### macOS Version

Sonoma 14.5

### Description

There are a few situations in which I've experienced the menu bar notifications list clearing and then repopulating a few seconds later. I've also experienced the menu entirely disappearing (even while I'm actively reading it) and then reappearing after a few seconds.

I'm guessing that this comes down to some kind of data caching issue, or maybe some sort of rate limiting? Regardless, in all the situations I've experienced this, there should always be some data already fetched which is still usable.

### Steps To Reproduce

There are multiple ways I've been able to reproduce this, including some I can't figure out exactly what is triggering it. But this is one I can reproduce consistently:

1) Make sure you have at least 2 unread notifications. More is ok.
2) Click one of the notifications in the menu bar app
3) Wait 10 seconds. Often will happen with a lot less, but this should make sure you are able to reproduce
4) Reopen the menu bar. The items in the list will be shown at first, then the list will clear for a second, and then will be repopulated again with the original data.

My extension settings in case you need these:
- Authenticated to GitHub via token (not oauth)
- Unread notifications command:
  - Always show: false
  - Show unread count: true
  - Background refresh: true

**Comment by thomaslombart** [comment]
Could you share a recording? It'll make it easier to see what's happening.

**Comment by asportnoy** [comment]
Unfortunately not (at the moment at least) as almost all of my notifications are from private repos and are confidential. I will see if I can get enough other notifications to get a recording but not sure if/when that will happen.

**Comment by asportnoy** [comment]
Managed to get a recording, here you go. Notice how the PRs are there at first, then disappear for a second, then reappear a second later.

https://github.com/raycast/extensions/assets/14863373/03a531a2-bd91-4a9a-ad08-7bfe8cb00c13

In a similar vein, when I first opened up the menu after not opening it for a while, there was no data for a few seconds even though it knew the count. Shouldn't it have the actual notification data prefetched?
